affordance_primitives/src/screw_execution.cpp

```diff
@@ -60,9 +60,34 @@ bool APScrewExecutor::getScrewTwist(AffordancePrimitive::Request& req, Affordanc
   Eigen::VectorXd eigen_twist_moving_frame =
       getAdjointMatrix(tfmsg_moving_to_task_frame.transform) * twistToVector(twist_in_task_frame.twist);
 
+  // Figure out estimated wrench
+  Wrench wrench_in_task_frame;
+  wrench_in_task_frame.force.x = req.task_impedance.trans_x * twist_in_task_frame.twist.linear.x;
```

Comment:
Can see you went with task impedance. Does this have to be set for every task? If so, could you point me to where this is set and later passed to this bool function?

Reply:
The task impedance is set in the [input service request](https://github.com/UTNuclearRobotics/affordance_primitives/blob/2211b97327bb78bd3bf11295c4cfb3a4e738b39f/affordance_primitive_msgs/srv/AffordancePrimitive.srv#L18-L19).

In theory it does have to be set every task, but the default value is all 0's which will just result in the expected wrench in the EE to be 0 as well. Basically - if whatever is calling this service wants to use the "force required at the EE to do this task", then they will have to set the impedance for the task